simulator/scheduler/scheduler.go

```diff
@@ -67,124 +64,60 @@ func NewSchedulerService(client clientset.Interface, restclientCfg *restclient.C
-	var dynInformerFactory dynamicinformer.DynamicSharedInformerFactory
-	if restConfig != nil {
-		dynClient := dynamic.NewForConfigOrDie(restConfig)
-		dynInformerFactory = dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynClient, 0, v1.NamespaceAll, nil)
+	if err := cli.ContainerRestart(ctx, id, container.StopOptions{}); err != nil {
+		return err
 	}
-	evtBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{
-		Interface: clientSet.EventsV1(),
-	})
-	evtBroadcaster.StartRecordingToSink(ctx.Done())
-
-	s.currentSchedulerCfg = versionedcfg.DeepCopy()
-
-	var err error
-	// Extender service must be initialized using `versionedcfg.Extenders` config which is not override for simulator (before calling OverrideExtendersCfgToSimulator()).
-	s.extenderService, err = extender.New(clientSet, versionedcfg.Extenders, s.sharedStore)
+	inspect, err := cli.ContainerInspect(ctx, id)
 	if err != nil {
-		return xerrors.Errorf("New extender service: %w", err)
+		return err
 	}
-
-	// Override the Extenders config so that the connection is directed to the simulator server.
-	extender.OverrideExtendersCfgToSimulator(versionedcfg, s.simulatorPort)
-
-	versioned, err := ConvertConfigurationForSimulator(versionedcfg)
-	if err != nil {
-		return xerrors.Errorf("convert scheduler config to apply: %w", err)
+	if inspect.State.Status != "running" {
+		return xerrors.Errorf("restart container status is not running")
 	}
+	return nil
+}
 
-	cfg, err := ConvertSchedulerConfigToInternalConfig(versioned)
+func (s *Service) RestartScheduler(cfg *configv1.KubeSchedulerConfiguration) error {
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
 	if err != nil {
-		return xerrors.Errorf("convert scheduler config to internal one: %w", err)
+		klog.Info("shutdown scheduler...")
+		return err
 	}
 
-	cfg, err = filterOutNonAllowedChangesOnCfg(cfg)
+	oldCfg, err := simulatorconfig.GetSchedulerCfg()
 	if err != nil {
-		return xerrors.Errorf("filter out non allowed changes: %w", err)
+		return xerrors.Errorf("read old scheduler.yaml: %w", err)
 	}
 
-	registry, err := plugin.NewRegistry(s.sharedStore, cfg, nil)
+	containers, err := cli.ContainerList(ctx, container.ListOptions{})
 	if err != nil {
-		return xerrors.Errorf("plugin registry: %w", err)
+		return err
 	}
-
-	if s.sharedStore != nil {
-		// Resister the event handler function to store the result stored in the sharedStore in pod.
-		if err := s.sharedStore.ResisterResultSavingToInformer(clientSet, ctx.Done()); err != nil {
-			return xerrors.Errorf("ResisterResultSavingToInformer of sharedStore: %w", err)
+	for _, c := range containers {
+		if c.Names[0] != "/simulator-scheduler" {
+			continue
 		}
+		if err := restartContainer(ctx, cli, c.ID, cfg); err != nil {
+			if err := restartContainer(ctx, cli, c.ID, oldCfg); err != nil {
+				return xerrors.Errorf("oldConfig restart failed: %w", err)
+			}
+		}
+		break
 	}
```

Comment:
Move L98-L111 into `restartContainer`.
I mean, `restartContainer` should list containers, look for the scheduler one, and restart it.